### PR TITLE
feat/enterpriseportal: define SubscriptionLicenseChecksService

### PR DESCRIPTION
**Pull request by bobheadxi**
This PR defines an Enterprise Portal version of the "license check" API that is currently bundled in Souregraph as a separate gRPC service that will be implemented in Enterprise Portal (in a PR I will stack on top of this one). The API schema is pretty much the same, except:

- removed the `error` field, since I'm pretty sure we can return a real error instead in a connectRPC implementation for the same effect.
- made `license_key` a parameter, as we already propagate the license key all over the place (pings, telemetry) - this is a fairly low-frequency check so there shouldn't be any bandwidth concerns or the like. For now, this parameter will support the existing format so we can forward checks in dotcom to EP, but I've set a target removal release

Existing API: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/internal/dotcom/productsubscription/license_check_handler.go?L133

Part of https://linear.app/sourcegraph/issue/CORE-227

## Test plan

CI

### CODE REVIEW

**Comment by bobheadxi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64396?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#64396** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64396?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#64253** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64253?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#64057](https://github.com/sourcegraph/sourcegraph/pull/64057) <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64057?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#64115](https://github.com/sourcegraph/sourcegraph/pull/64115) <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64115?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#63959** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63959?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64114** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64114?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#63927** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/63927?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64087** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64087?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64317** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64317?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64090** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64090?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#64376** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/64376?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/enterpriseportal/subscriptionlicensechecks/v1/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,45 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@rules_proto_grpc//doc:defs.bzl", "doc_template_compile")
+load("@rules_buf//buf:defs.bzl", "buf_lint_test")
+
+# Bazel-generated files are different from what is generated locally by buf and
```
Comment (eseliger) [inline lib/enterpriseportal/subscriptionlicensechecks/v1/BUILD.bazel:6]: Huh how’s that 🙃

**Review by eseliger — APPROVED** [review]

File: lib/enterpriseportal/subscriptionlicensechecks/v1/BUILD.bazel (hunk 1 above)
Comment (bobheadxi) [reply lib/enterpriseportal/subscriptionlicensechecks/v1/BUILD.bazel:6]: the issue mentioned in the docstring https://github.com/sourcegraph/devx-support/issues/932 has an archive of my debugging efforts. tl;dr i'm not sure, and the bazel-generated files are invalid, causing everything to fail to build if you use Bazel but not if you use Go

**Comment by bobheadxi** [comment]
Discussed with @eseliger offline, I'm introducing another change here:

> made license_key a parameter, as we already propagate the license key all over the place (pings, telemetry) - this is a fairly low-frequency check so there shouldn't be any bandwidth concerns or the like.

For now, this parameter will support the existing format so we can forward checks in dotcom to EP, but I've set a target removal release

**Review by unknwon — APPROVED** [review]

**Pull request merged**